### PR TITLE
Don't use raw values for strings unless it's in a flow expression

### PR DESCRIPTION
**Pull request by Janpot**
#4109

### CODE REVIEW

File: packages/babel-generator/src/printer.js [hunk 1]
```diff
@@ -110,9 +110,13 @@ export default class Printer extends Buffer {
     if (!this.format.minified) {
       let extra = this.getPossibleRaw(node);
       if (extra) {
-        this.push("");
-        this._push(extra);
-        return;
+        let isString = extra.indexOf("\"") >= 0 || extra.indexOf("'") >= 0;
+        let isFlow = parent && t.isFlow(parent);
+        if (!isString || isFlow) {
```
Comment (loganfsmyth) [inline packages/babel-generator/src/printer.js]: I'm not that familiar with Flow, what's the driving factor for checking `isFlow` in this case?

Comment (Janpot) [inline packages/babel-generator/src/printer.js]: for me personally, nothing. But it failed the tests otherwise.

**Comment by loganfsmyth** [comment]
@amasad Looks like this whole block was added in https://github.com/babel/babel/commit/4b99d18ea79c139ee2fd0641fe44cf60daf1f6cb and I can't tell why. Was performance your concern? Not clear on why we'd favor raw values in the non-minified case.

Seems nice for number formatting, but otherwise unnecessary.

**Comment by amasad** [comment]
>@amasad Looks like this whole block was added in 4b99d18 and I can't tell why. Was performance your concern? Not clear on why we'd favor raw values in the non-minified case.

That diff is confusing. Look at the PR: https://github.com/babel/babel/pull/3145
Initially, I reprinted always, but @jmm raised a valid concern on the diff.

**Comment by hzoo** [comment]
We are removing the quotes option from babel-generator (https://github.com/babel/babel/issues/4109) so closing this, thanks for the PR!

**Pull request closed without merging**